### PR TITLE
Migrate old Azure SDKs into to use `github.com/Azure/azure-sdk-for-go/sdk/azidentity`

### PR DESCRIPTION
**Pull request by JorTurFer**
This PR has migrated some old Azure SDKs into the (unified) new one 

- EventHub: `github.com/Azure/azure-event-hubs-go/v3` -> `github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs`
- Blob Storage: `github.com/Azure/azure-storage-blob-go` -> `github.com/Azure/azure-sdk-for-go/sdk/storage/azblob`
- Storage queues: `github.com/Azure/azure-storage-queue-go` -> `github.com/Azure/azure-sdk-for-go/sdk/storage/azqueue`
- Azure Key Vault: `github.com/Azure/azure-sdk-for-go` -> `github.com/Azure/azure-sdk-for-go/sdk/security/keyvault/azsecrets`
- Azure Monitor: `github.com/Azure/azure-sdk-for-go` -> `github.com/Azure/azure-sdk-for-go/sdk/monitor/azquery` 
- Azure Log Analytics:  `no-sdk` -> `github.com/Azure/azure-sdk-for-go/sdk/monitor/azquery`
- Azure Data Explorer: `github.com/Azure/azure-kusto-go/kusto` uses `github.com/Azure/azure-sdk-for-go/sdk/azidentity`
- Azure Application Insights: Currently it's not supported, [I've opened and issue asking for it](https://github.com/Azure/azure-sdk-for-go/issues/22355) for migrating it in the future

Additionally to improve the testing experience, I've replaced `go test ... | junit....` with `gotestsum` -> https://github.com/gotestyourself/gotestsum

### Checklist

- [x] When introducing a new scaler, I agree with the [scaling governance policy](https://github.com/kedacore/governance/blob/main/SCALERS.md)
- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)* https://github.com/kedacore/keda-docs/pull/1313
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes https://github.com/kedacore/keda/issues/5470

### CODE REVIEW

File: pkg/scalers/azure_data_explorer_scaler_test.go [hunk 1]
```diff
@@ -40,13 +40,14 @@ type dataExplorerMetricIdentifier struct {
...
-	dataExplorerThreshold = "1"
-	dataExplorerEndpoint  = "https://test-keda-e2e.eastus.kusto.windows.net"
+	aadAppClientID          = "eebdbbab-cf74-4791-a5c6-1ef5d90b1fa8"
+	aadAppSecret            = "test_app_secret"
+	activeDirectoryEndpoint = "activeDirectoryEndpoint"
+	azureTenantID           = "8fe57c22-02b1-4b87-8c24-ae21dea4fa6a"
+	databaseName            = "test_database"
+	dataExplorerQuery       = "print 3"
+	dataExplorerThreshold   = "1"
+	dataExplorerEndpoint    = "https://test-keda-e2e.eastus.kusto.windows.net"
```
Comment (JorTurFer) [inline pkg/scalers/azure_data_explorer_scaler_test.go:50]: The change here is that I've moved `activeDirectoryEndpoint` from eventhub_scaler_test to here as it's not needed anymore there

File: pkg/scalers/azure_eventhub_scaler.go [hunk 2]
```diff
@@ -155,17 +155,6 @@ func parseCommonAzureEventHubMetadata(config *scalersconfig.ScalerConfig, meta *
...
-	meta.eventHubInfo.EventHubResourceURL = azure.DefaultEventhubResourceURL
-	if val, ok := config.TriggerMetadata["cloud"]; ok {
-		if strings.EqualFold(val, azure.PrivateCloud) {
-			if resourceURL, ok := config.TriggerMetadata["eventHubResourceURL"]; ok {
-				meta.eventHubInfo.EventHubResourceURL = resourceURL
-			} else {
-				return fmt.Errorf("eventHubResourceURL must be provided for %s cloud type", azure.PrivateCloud)
-			}
-		}
-	}
```
Comment (JorTurFer) [inline pkg/scalers/azure_eventhub_scaler.go:167]: I haven't been able to find where is this used in the new SDK. From my understanding, this was used for requesting the tokens (we requested them using http client) and now the SDK manages it in a better way

File: pkg/scalers/azure_eventhub_scaler.go [hunk 3]
```diff
@@ -175,12 +164,6 @@ func parseCommonAzureEventHubMetadata(config *scalersconfig.ScalerConfig, meta *
 	}
 	meta.eventHubInfo.ServiceBusEndpointSuffix = serviceBusEndpointSuffix
 
-	activeDirectoryEndpoint, err := azure.ParseActiveDirectoryEndpoint(config.TriggerMetadata)
-	if err != nil {
-		return err
-	}
-	meta.eventHubInfo.ActiveDirectoryEndpoint = activeDirectoryEndpoint
```
Comment (JorTurFer) [inline pkg/scalers/azure_eventhub_scaler.go:182]: Same as above, I think that this is now managed by the SDK. When we used rest client we needed to provide more info

**Review by zroubalik — APPROVED** [review]

**Review by zroubalik — COMMENTED** [review]
@JorTurFer what is the status on this please?

**Comment by JorTurFer** [comment]
It's ready to merge from code pov (just pending to rebase main branch). But I think that waiting until after v2.14 is released could be a good idea because I'm not 100% sure if I have dropped some AAD-Pod-Identity support and taking into account that v2.15 will remove AAD-Pod-Identity I'm more confident adding this after removing AAD-Pod-Identity.

**Comment by zroubalik** [comment]
> It's ready to merge from code pov (just pending to rebase main branch). But I think that waiting until after v2.14 is released could be a good idea because I'm not 100% sure if I have dropped some AAD-Pod-Identity support and taking into account that v2.15 will remove AAD-Pod-Identity I'm more confident adding this after removing AAD-Pod-Identity.

Okay, sounds good!

**Comment by JorTurFer** [comment]
time to rebase :)

**Review by zroubalik — APPROVED** [review]

**Pull request merged**